urllib3/exceptions.py

```diff
@@ -112,6 +112,9 @@ class ConnectTimeoutError(TimeoutError):
     "Raised when a socket timeout occurs while connecting to a server"
     pass
 
+class NewConnectionError(ConnectTimeoutError, PoolError):
```

Comment:
Do these base exceptions make sense?

The old behaviour was partly encompassed by ConnectTimeoutError before (not completely), whereas now it's all under NewConnectionError so I think it's sensible to make it backwards-compatible.

Otherwise, not sure what is should go under—PoolError? Something else?